Assets/Scripts/Combat/CombatCamera.cs

```diff
@@ -4,16 +4,49 @@
 public class CombatCamera : MonoBehaviour
 {
 
+    const int DIR_NORTH = 0;
+    const int DIR_EAST = 1;
+    const int DIR_SOUTH = 2;
+    const int DIR_WEST = 3;
+
+    int dir = 0;
+    bool rotationFrozen = false;
+
     private float updateTimeSeconds = 0.25f;
 
+    const float MIN_SCROLL = -2.5f;
+    const float MAX_SCROLL = 2.5f;
+
     public void RotateLeft()
     {
-        transform.Rotate(Vector3.up, 90, Space.Self);
+        if (rotationFrozen) return;
+        StartCoroutine(GradualizeRotation(90, .025f));
+        dir = (dir + 3) % 4;
     }
 
     public void RotateRight()
     {
-        transform.Rotate(Vector3.up, -90, Space.Self);
+        if (rotationFrozen) return;
+        StartCoroutine(GradualizeRotation(-90, .025f));
+        dir = (dir + 1) % 4;
+    }
+
+    void CameraSpeedCalc(ref Vector3 cameraSpeed, int edge)
```

Comment:
If you don't use the ref keyword, does that mean you won't change the cameraSpeed variable in MouseScroll()?

Reply:
Right. In C#, you can pass-by-value or pass-by-reference.

In pass-by-value, it makes a copy of the variable with the same value.

In pass-by-reference it passes a reference to the actual variable itself, so any changes applied to that variable will carry over into the calling function.

Pass-by-value is default for most simple types including Vector3, but there are some types that are considered reference types (such as objects, Unity GameObjects, etc.) that are passed by reference by default.

Pass-by-reference is great for situations like this where you want to mutate a variable or munge some data. It can also be handy when you have a variable that takes up a bunch of memory that you are going to need to pass around a lot, since avoiding copying it can make your program run faster.